src/main/java/org/spongepowered/common/mixin/core/world/extent/MixinExtent.java

```diff
@@ -163,6 +163,26 @@ public Location getFullBlock(int x, int y, int z) {
     }
 
     @Override
+    public byte getLuminance(Vector3i position) {
+        return getLuminance(position.getX(), position.getY(), position.getZ());
+    }
+
+    @Override
+    public byte getLuminance(int x, int y, int z) {
+        return (byte) Math.max(getLuminanceFromGround(x, y, z), getLuminanceFromSky(x, y, z));
```

Comment:
Similarly, NMS lacks a sufficient server side function to provide the light level present at a block, thus, the higher of the two values is used, as per Minecraft's current behavior.